djvm/src/main/kotlin/net/corda/djvm/rules/implementation/ReturnTypeWrapper.kt

```diff
@@ -1,27 +1,73 @@
 package net.corda.djvm.rules.implementation
 
+import net.corda.djvm.code.EMIT_AFTER_INVOKE
 import net.corda.djvm.code.Emitter
 import net.corda.djvm.code.EmitterContext
 import net.corda.djvm.code.Instruction
 import net.corda.djvm.code.instructions.MemberAccessInstruction
+import org.objectweb.asm.Opcodes.INVOKESPECIAL
+import org.objectweb.asm.Opcodes.INVOKESTATIC
+import org.objectweb.asm.Opcodes.INVOKEVIRTUAL
 
 /**
  * Whitelisted classes may still return [java.lang.String] from some
  * functions, e.g. [java.lang.Object.toString]. So always explicitly
  * invoke [sandbox.java.lang.String.toDJVM] after these.
+ *
+ * These factory functions also need special handling:
+ *   [java.util.concurrent.atomic.AtomicIntegerFieldUpdater.newUpdater]
```

Comment:
What is special about `AtomicIntegerFieldUpdater`? I am missing the connection between this class and `Currency`

Reply:
The `AtomicLongFieldUpdater` is used by `Currency`. All _3_ `Atomic*FieldUpdater` classes are used by the Tokens SDK / Kotlin (which was my use-case for testing `Currency`).